app/controllers/kudos_controller.rb

```diff
@@ -20,15 +20,22 @@ def edit
     redirect_to kudos_path, notice: 'You are not owner of this kudo.' if @kudo.giver_id != current_employee.id
   end
 
-  # POST /kudos
   def create
-    @kudo = Kudo.new(kudo_params)
-    @kudo.giver = current_employee
-
-    if @kudo.save
-      redirect_to kudos_url, notice: 'Kudo was successfully created.'
+    if current_employee.number_of_available_kudos.zero?
+      redirect_to kudos_path, notice: 'Kudo was not  created. Your number of available kudos is 0'
     else
-      render :new
+      @kudo = Kudo.new(kudo_params)
+      @kudo.giver = current_employee
+
+      if @kudo.save
+        @kudo.giver.number_of_available_kudos -= 1
+        @kudo.giver.save
```

Comment:
what happen if we get an error while `@kudo.giver.save`
I would suggest using transaction here
https://longliveruby.com/articles/introduction-to-rails-transactions